apps/condo/domains/common/components/tasks/storage/TasksLocalStorage.tsx

```diff
@@ -0,0 +1,126 @@
+import find from 'lodash/find'
+import findIndex from 'lodash/findIndex'
+import { ITasksStorage, OnCompleteFunc } from '../index'
+
+const LOCAL_STORAGE_TASKS_KEY = 'tasks'
+
+// TODO(antonal): implement polling a task record from browser's Local Storage
+const StopPollingStub = () => {
+    return
+}
+
+const isServerSide = typeof window === 'undefined'
+
+/**
+ * Used to store third-party task records, unknown for Condo API.
+ * Third-party task records will come from mini-apps
+ * On first page load all locally stored tasks can be fetched to display
+ * them in progress panel
+ * Uses browser `localstorage`
+ * NOTE: not working with SSR because there is no `window` object
+ * TODO(antonal): load tasks created by current user only
+ */
+export class TasksLocalStorage implements ITasksStorage {
+
+    useTasks ({ status }) {
+        if (isServerSide) {
+            // Gracefully return empty results in SSR mode, no need to throw errors or do something extra
+            return { records: [] }
+        }
+        const tasks = this.getAllItemsFromStorage()
+        if (!status) {
+            return tasks
+        }
+        const records = tasks.filter(task => task.status === status)
+        return { records }
```

Comment:
looks like this function returns either array or object, this could be confusing.